models/sla_policy/reply_time/int_zendesk__reply_time_business_hours.sql

```diff
@@ -48,20 +77,53 @@ with ticket_schedules as (
     on ticket_schedules.schedule_id = schedule_business_hours.schedule_id
   where sla_policy_applied.in_business_hours
     and metric in ('next_reply_time', 'first_reply_time')
-  
+
+), first_reply_solve_times as (
+  select
+    ticket_sla_applied_with_schedules.ticket_id,
+    ticket_sla_applied_with_schedules.ticket_created_at,
+    ticket_sla_applied_with_schedules.valid_starting_at,
+    ticket_sla_applied_with_schedules.ticket_current_status,
+    ticket_sla_applied_with_schedules.metric,
+    ticket_sla_applied_with_schedules.latest_sla,
+    ticket_sla_applied_with_schedules.sla_applied_at,
+    ticket_sla_applied_with_schedules.target,
+    ticket_sla_applied_with_schedules.in_business_hours,
+    ticket_sla_applied_with_schedules.sla_policy_name,
+    ticket_sla_applied_with_schedules.schedule_id,
+    ticket_sla_applied_with_schedules.start_time_in_minutes_from_week,
+    ticket_sla_applied_with_schedules.total_schedule_weekly_business_minutes,
+    ticket_sla_applied_with_schedules.start_week_date,
+    min(reply_time.reply_at) as first_reply_time,
+    min(ticket_solved_times.solved_at) as first_solved_time
+  from ticket_sla_applied_with_schedules
+  left join reply_time
+    on reply_time.ticket_id = ticket_sla_applied_with_schedules.ticket_id
+    and reply_time.reply_at > ticket_sla_applied_with_schedules.sla_applied_at
+  left join ticket_solved_times
+    on ticket_sla_applied_with_schedules.ticket_id = ticket_solved_times.ticket_id
+    and ticket_solved_times.solved_at > ticket_sla_applied_with_schedules.sla_applied_at
+  {{ dbt_utils.group_by(n=14) }}
+
+), week_index_calc as (
+    select 
+        *,
+        {{ dbt.datediff("sla_applied_at", "least(first_reply_time, first_solved_time)", "week") }} + 1 as week_index
```

Comment:
When doing validations I found that this code produces faulty results if either `first_reply_time` **OR** `first_solved_time` are `null`. This is due the the greatest/least functions returning `null` if either field being compared is `null` (see details [here](https://cloud.google.com/bigquery/docs/reference/standard-sql/mathematical_functions#least)). Therefore, we should be performing a coalesce and defaulting to the current_timestamp if the field is null. This will ensure we don't return null and filter out any records erroneously.
```suggestion
        {{ dbt.datediff("sla_applied_at", "least(coalesce(first_reply_time, " ~ dbt.current_timestamp() ~ "), coalesce(first_solved_time, " ~ dbt.current_timestamp() ~ "))", "week") }} + 1 as week_index
```

Reply:
Makes sense-- so if both `first_reply_time` and `first_reply_time` are null, then the date diff will just be comparing `sla_applied_at` and current_timestamp.

Reply:
Correct!